src/main/scala/sparkz/core/network/NetworkController.scala

```diff
@@ -209,6 +213,10 @@ class NetworkController(settings: NetworkSettings,
     }
   }
 
+  private def updateLastConnectionAttemptTimestamp(remoteAddress: InetSocketAddress): Unit = {
+    peersLastConnectionAttempts += remoteAddress -> sparkzContext.timeProvider.time()
```

Comment:
I see that we never remove entries from this map so theoretically may be overflown as it was with peers map before.
Let's set the limit and cleanup the oldest entry in case we reach the max size.